nebula_ros/src/continental/continental_ars548_decoder_ros_wrapper.cpp

```diff
@@ -0,0 +1,851 @@
+                           std::cos(yaw) * half_length * reference_to_center_[reference_index][0] -
+                           std::sin(yaw) * half_width * reference_to_center_[reference_index][1];
+    track_msg.position.y = object.position.y +
+                           std::sin(yaw) * half_length * reference_to_center_[reference_index][0] +
+                           std::cos(yaw) * half_width * reference_to_center_[reference_index][1];
+    track_msg.position.z = object.position.z;
+
+    track_msg.velocity = object.absolute_velocity;
+    track_msg.acceleration = object.absolute_acceleration;
+    track_msg.size.x = object.shape_length_edge_mean;
+    track_msg.size.y = object.shape_width_edge_mean;
+    track_msg.size.z = 1.f;
+
+    uint8_t max_score = object.classification_unknown;
+    track_msg.classification = UNKNOWN_ID;
+
+    if (object.classification_car > max_score) {
+      max_score = object.classification_car;
+      track_msg.classification = CAR_ID;
+    }
+    if (object.classification_truck > max_score) {
+      max_score = object.classification_truck;
+      track_msg.classification = TRUCK_ID;
+    }
+    if (object.classification_motorcycle > max_score) {
+      max_score = object.classification_motorcycle;
+      track_msg.classification = MOTORCYCLE_ID;
+    }
+    if (object.classification_bicycle > max_score) {
+      max_score = object.classification_bicycle;
+      track_msg.classification = BICYCLE_ID;
+    }
+    if (object.classification_pedestrian > max_score) {
+      max_score = object.classification_pedestrian;
+      track_msg.classification = PEDESTRIAN_ID;
+    }
+
+    track_msg.position_covariance[0] = static_cast<float>(object.position_std.x);
+    track_msg.position_covariance[1] = object.position_covariance_xy;
+    track_msg.position_covariance[2] = 0.f;
+    track_msg.position_covariance[3] = static_cast<float>(object.position_std.y);
+    track_msg.position_covariance[4] = 0.f;
+    track_msg.position_covariance[5] = static_cast<float>(object.position_std.z);
+
+    track_msg.velocity_covariance[0] = static_cast<float>(object.absolute_velocity_std.x);
+    track_msg.velocity_covariance[1] = object.absolute_velocity_covariance_xy;
+    track_msg.velocity_covariance[2] = 0.f;
+    track_msg.velocity_covariance[3] = static_cast<float>(object.absolute_velocity_std.y);
+    track_msg.velocity_covariance[4] = 0.f;
+    track_msg.velocity_covariance[5] = static_cast<float>(object.absolute_velocity_std.z);
+
+    track_msg.acceleration_covariance[0] = static_cast<float>(object.absolute_acceleration_std.x);
+    track_msg.acceleration_covariance[1] = object.absolute_acceleration_covariance_xy;
+    track_msg.acceleration_covariance[2] = 0.f;
+    track_msg.acceleration_covariance[3] = static_cast<float>(object.absolute_acceleration_std.y);
+    track_msg.acceleration_covariance[4] = 0.f;
+    track_msg.acceleration_covariance[5] = static_cast<float>(object.absolute_acceleration_std.z);
```

Comment:
Going by the variable names `_std` etc. you are populating the covariance matrix with standard deviations in place of variances, is this intentional?